### ISSUE TITLE
Piecharts with overlapping labels

### ISSUE THREAD

**Issue by DoubleT13** [useblocks/sphinxcontrib-needs#498]
With contents containing no data (or datasets where one content is a lot bigger than the others) , the labels of a `needpie::` are overlapping.

**Comment by twodrops** [comment]
Attaching an example
![needpie](https://user-images.githubusercontent.com/103930/154531325-64a4f5be-25b7-4d70-9503-f4cef9296aeb.png)

**Comment by danwos** [comment]
I agree that this is not nice, but I see no easy way to fix it inside Sphinx-Needs, as the user is more or less configuring matplotlib library and Sphinx-Needs has no understanding of the needed final result.

We could throw away sections, which have 0 need objects.
But I also see use cases, where this information is needed.

I think this is the responsibility of the author to define some working configuration for the filtered data.
For sure if there is a matplotlib feature to somehow support this, I'm open to add the needed configuration.

I would like to close this issue :)

**Comment by DoubleT13** [comment]
What about implementing a workaround by having all relevant information (label + relative number + absolute number) in the legend and remove the labels from the pie chart itself. (similar to [here](https://stackoverflow.com/questions/23577505/how-to-avoid-overlapping-of-labels-autopct-in-a-matplotlib-pie-chart))

This could either be the default behavior of `:legend:` option or a new option `:legend_with_data:` (or a more compact name).
Removing the labels from the plot when using the legend would make more sense, anyway.

What do you think?

**Comment by danwos** [comment]
Please have a look into https://sphinxcontrib-needs.readthedocs.io/en/latest/directives/needpie.html#overlapping-labels for the new implementation. I like it a lot. And thanks to @PhilipPartsch for implementing this.
